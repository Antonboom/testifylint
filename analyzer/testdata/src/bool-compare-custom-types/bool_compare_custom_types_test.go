package boolcomparecustomtypes_test

import (
	"testing"

	"bool-compare-custom-types/types"
	"github.com/stretchr/testify/assert"
)

type MyBool bool

func TestBoolCompareChecker_CustomTypes(t *testing.T) {
	var b MyBool
	{
		assert.Equal(t, false, b)       // want "bool-compare: use assert\\.False"
		assert.EqualValues(t, false, b) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, b)

		assert.Equal(t, true, b)       // want "bool-compare: use assert\\.True"
		assert.EqualValues(t, true, b) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, b)

		assert.NotEqual(t, false, b)       // want "bool-compare: use assert\\.True"
		assert.NotEqualValues(t, false, b) // want "bool-compare: use assert\\.True"

		assert.NotEqual(t, true, b)       // want "bool-compare: use assert\\.False"
		assert.NotEqualValues(t, true, b) // want "bool-compare: use assert\\.False"

		assert.True(t, b == true)  // want "bool-compare: need to simplify the assertion"
		assert.True(t, b != false) // want "bool-compare: need to simplify the assertion"
		assert.True(t, b == false) // want "bool-compare: use assert\\.False"
		assert.True(t, b != true)  // want "bool-compare: use assert\\.False"

		assert.False(t, b == true)  // want "bool-compare: need to simplify the assertion"
		assert.False(t, b != false) // want "bool-compare: need to simplify the assertion"
		assert.False(t, b == false) // want "bool-compare: use assert\\.True"
		assert.False(t, b != true)  // want "bool-compare: use assert\\.True"
	}

	var extB types.Bool
	{
		assert.Equal(t, false, extB)       // want "bool-compare: use assert\\.False"
		assert.EqualValues(t, false, extB) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, extB)

		assert.Equal(t, true, extB)       // want "bool-compare: use assert\\.True"
		assert.EqualValues(t, true, extB) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, extB)

		assert.NotEqual(t, false, extB)       // want "bool-compare: use assert\\.True"
		assert.NotEqualValues(t, false, extB) // want "bool-compare: use assert\\.True"

		assert.NotEqual(t, true, extB)       // want "bool-compare: use assert\\.False"
		assert.NotEqualValues(t, true, extB) // want "bool-compare: use assert\\.False"

		assert.True(t, extB == true)  // want "bool-compare: need to simplify the assertion"
		assert.True(t, extB != false) // want "bool-compare: need to simplify the assertion"
		assert.True(t, extB == false) // want "bool-compare: use assert\\.False"
		assert.True(t, extB != true)  // want "bool-compare: use assert\\.False"

		assert.False(t, extB == true)  // want "bool-compare: need to simplify the assertion"
		assert.False(t, extB != false) // want "bool-compare: need to simplify the assertion"
		assert.False(t, extB == false) // want "bool-compare: use assert\\.True"
		assert.False(t, extB != true)  // want "bool-compare: use assert\\.True"
	}

	var extSuperB types.SuperBool
	{
		assert.Equal(t, false, extSuperB)       // want "bool-compare: use assert\\.False"
		assert.EqualValues(t, false, extSuperB) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, extSuperB)

		assert.Equal(t, true, extSuperB)       // want "bool-compare: use assert\\.True"
		assert.EqualValues(t, true, extSuperB) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, extSuperB)

		assert.NotEqual(t, false, extSuperB)       // want "bool-compare: use assert\\.True"
		assert.NotEqualValues(t, false, extSuperB) // want "bool-compare: use assert\\.True"

		assert.NotEqual(t, true, extSuperB)       // want "bool-compare: use assert\\.False"
		assert.NotEqualValues(t, true, extSuperB) // want "bool-compare: use assert\\.False"

		assert.True(t, extSuperB == true)  // want "bool-compare: need to simplify the assertion"
		assert.True(t, extSuperB != false) // want "bool-compare: need to simplify the assertion"
		assert.True(t, extSuperB == false) // want "bool-compare: use assert\\.False"
		assert.True(t, extSuperB != true)  // want "bool-compare: use assert\\.False"

		assert.False(t, extSuperB == true)  // want "bool-compare: need to simplify the assertion"
		assert.False(t, extSuperB != false) // want "bool-compare: need to simplify the assertion"
		assert.False(t, extSuperB == false) // want "bool-compare: use assert\\.True"
		assert.False(t, extSuperB != true)  // want "bool-compare: use assert\\.True"
	}

	// Crazy cases:
	{
		assert.Equal(t, true, types.Bool(extSuperB))    // want "bool-compare: use assert\\.True"
		assert.Equal(t, true, types.SuperBool(b))       // want "bool-compare: use assert\\.True"
		assert.Equal(t, true, bool(types.SuperBool(b))) // want "bool-compare: use assert\\.True"
		assert.True(t, !bool(types.SuperBool(b)))       // want "bool-compare: use assert\\.False"
		assert.False(t, !bool(types.SuperBool(b)))      // want "bool-compare: use assert\\.True"
	}
}

func TestBoolCompareChecker_CustomTypes_Format(t *testing.T) {
	var predicate MyBool
	assert.Equal(t, true, predicate)                                   // want "bool-compare: use assert\\.True"
	assert.Equal(t, true, predicate, "msg")                            // want "bool-compare: use assert\\.True"
	assert.Equal(t, true, predicate, "msg with arg %d", 42)            // want "bool-compare: use assert\\.True"
	assert.Equal(t, true, predicate, "msg with args %d %s", 42, "42")  // want "bool-compare: use assert\\.True"
	assert.Equalf(t, true, predicate, "msg")                           // want "bool-compare: use assert\\.Truef"
	assert.Equalf(t, true, predicate, "msg with arg %d", 42)           // want "bool-compare: use assert\\.Truef"
	assert.Equalf(t, true, predicate, "msg with args %d %s", 42, "42") // want "bool-compare: use assert\\.Truef"
}
