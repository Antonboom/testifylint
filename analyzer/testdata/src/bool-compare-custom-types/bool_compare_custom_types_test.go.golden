package boolcomparecustomtypes_test

import (
	"testing"

	"bool-compare-custom-types/types"
	"github.com/stretchr/testify/assert"
)

type MyBool bool

func TestBoolCompareChecker_CustomTypes(t *testing.T) {
	var b MyBool
	{
		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, b)

		assert.True(t, bool(b)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(b)) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, b)

		assert.True(t, bool(b)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(b)) // want "bool-compare: use assert\\.True"

		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"

		assert.True(t, bool(b))  // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(b))  // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(b)) // want "bool-compare: use assert\\.False"

		assert.False(t, bool(b)) // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(b)) // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(b))  // want "bool-compare: use assert\\.True"
		assert.True(t, bool(b))  // want "bool-compare: use assert\\.True"
	}

	var extB types.Bool
	{
		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, extB)

		assert.True(t, bool(extB)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extB)) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, extB)

		assert.True(t, bool(extB)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extB)) // want "bool-compare: use assert\\.True"

		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"

		assert.True(t, bool(extB))  // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(extB))  // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extB)) // want "bool-compare: use assert\\.False"

		assert.False(t, bool(extB)) // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(extB)) // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(extB))  // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extB))  // want "bool-compare: use assert\\.True"
	}

	var extSuperB types.SuperBool
	{
		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"
		assert.Exactly(t, false, extSuperB)

		assert.True(t, bool(extSuperB)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extSuperB)) // want "bool-compare: use assert\\.True"
		assert.Exactly(t, true, extSuperB)

		assert.True(t, bool(extSuperB)) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extSuperB)) // want "bool-compare: use assert\\.True"

		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"

		assert.True(t, bool(extSuperB))  // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(extSuperB))  // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"
		assert.False(t, bool(extSuperB)) // want "bool-compare: use assert\\.False"

		assert.False(t, bool(extSuperB)) // want "bool-compare: need to simplify the assertion"
		assert.False(t, bool(extSuperB)) // want "bool-compare: need to simplify the assertion"
		assert.True(t, bool(extSuperB))  // want "bool-compare: use assert\\.True"
		assert.True(t, bool(extSuperB))  // want "bool-compare: use assert\\.True"
	}

	// Crazy cases:
	{
		assert.True(t, bool(types.Bool(extSuperB))) // want "bool-compare: use assert\\.True"
		assert.True(t, bool(types.SuperBool(b)))    // want "bool-compare: use assert\\.True"
		assert.True(t, bool(types.SuperBool(b)))    // want "bool-compare: use assert\\.True"
		assert.False(t, bool(types.SuperBool(b)))   // want "bool-compare: use assert\\.False"
		assert.True(t, bool(types.SuperBool(b)))    // want "bool-compare: use assert\\.True"
	}
}

func TestBoolCompareChecker_CustomTypes_Format(t *testing.T) {
	var predicate MyBool
	assert.True(t, bool(predicate))                                   // want "bool-compare: use assert\\.True"
	assert.True(t, bool(predicate), "msg")                            // want "bool-compare: use assert\\.True"
	assert.True(t, bool(predicate), "msg with arg %d", 42)            // want "bool-compare: use assert\\.True"
	assert.True(t, bool(predicate), "msg with args %d %s", 42, "42")  // want "bool-compare: use assert\\.True"
	assert.Truef(t, bool(predicate), "msg")                           // want "bool-compare: use assert\\.Truef"
	assert.Truef(t, bool(predicate), "msg with arg %d", 42)           // want "bool-compare: use assert\\.Truef"
	assert.Truef(t, bool(predicate), "msg with args %d %s", 42, "42") // want "bool-compare: use assert\\.Truef"
}
